pinot-core/src/main/java/com/linkedin/pinot/core/startree/v2/AggregationFunctionColumnPairBuffer.java

```diff
@@ -0,0 +1,191 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.startree.v2;
+
+import java.util.List;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import com.linkedin.pinot.common.data.FieldSpec;
+import com.linkedin.pinot.core.common.datatable.ObjectType;
+import com.linkedin.pinot.core.segment.memory.PinotDataBuffer;
+import com.linkedin.pinot.core.common.datatable.ObjectCustomSerDe;
+
+
+public class AggregationFunctionColumnPairBuffer {
+
+  int _totalBytes;
+  protected final Object[] _values;
+  private final List<AggregationFunctionColumnPair> _aggFunColumnPairs;
+
+  public AggregationFunctionColumnPairBuffer(Object[] values, List<AggregationFunctionColumnPair> aggFunColumnPairs) {
+    _values = values;
+    _aggFunColumnPairs = aggFunColumnPairs;
+
+    calculateBytesRequired();
+  }
+
+  /**
+   * NOTE: pass in byte size for performance. Byte size can be calculated by adding up field size for all metrics.
+   */
+  public byte[] toBytes() {
+    byte[] bytes = new byte[_totalBytes];
+    ByteBuffer buffer = ByteBuffer.wrap(bytes).order(PinotDataBuffer.NATIVE_ORDER);
+
+    for (int i = 0; i < _aggFunColumnPairs.size(); i++) {
+      AggregationFunctionColumnPair pair = _aggFunColumnPairs.get(i);
+      AggregationFunction factory = AggregationFunctionFactory.getAggregationFunction(pair.getFunctionType().getName());
+
+      switch (factory.getResultDataType()) {
+        case INT:
+          buffer.putInt(new Integer(_values[i].toString()));
+          break;
+        case LONG:
+          buffer.putLong(new Long(_values[i].toString()));
+          break;
+        case FLOAT:
+          buffer.putFloat(new Float(_values[i].toString()));
+          break;
+        case DOUBLE:
+          buffer.putDouble(new Double(_values[i].toString()));
+          break;
+        case BYTES:
+          int size = 0;
+          try {
+            size = getObjectSize(factory, _values[i]);
+          } catch (IOException e) {
+            e.printStackTrace();
+          }
+          buffer.putInt(size);
+          try {
+            buffer.put(ObjectCustomSerDe.serialize(_values[i]));
+          } catch (IOException e) {
+            e.printStackTrace();
+          }
+          break;
+        default:
+          throw new IllegalStateException();
+      }
+    }
+
+    return bytes;
+  }
+
+  /**
+   * NOTE: pass in byte size for performance. Byte size can be calculated by adding up field size for all metrics.
```

Comment:
This comment does not apply to the method

Reply:
done